Sources/Helpers/DirectoryObserver.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  DirectoryObserver.swift
+//  OsmAnd Maps
+//
+//  Created by Skalii on 26.07.2024.
+//  Copyright © 2024 OsmAnd. All rights reserved.
+//
+
+import Foundation
+
+@objc enum DirectoryObserverType: Int {
```

Comment:
DirectoryObserver should be path based, not specific type. Drop DirectoryObserverType.